src/partition/repart.c

```diff
@@ -3387,7 +3448,11 @@ static int context_mkfs(Context *context) {
                 if (r < 0)
                         return r;
 
-                r = make_filesystem(fsdev, p->format, strempty(p->new_label), NULL, fs_uuid, arg_discard);
+                r = partition_populate_directory(p, &root, &tmp_root);
```

Comment:
maybe add a comment somewhere here clarifying that we so far generally prefer to copy things in ourselves, to better support xattrs, chattr flags and similar, but for some file systems that cannot work.